src/tools/rbd_mirror/group_replayer/Replayer.cc

```diff
@@ -554,30 +562,128 @@ void Replayer<I>::handle_load_remote_group_snapshots(int r) {
       ++remote_snap;
     }
   }
+  is_resync_requested();
+}
+
+template <typename I>
+void Replayer<I>::is_resync_requested() {
+  dout(10) << "m_local_group_id=" << m_local_group_id << dendl;
+
+  librados::ObjectReadOperation op;
+  librbd::cls_client::metadata_get_start(&op, RBD_GROUP_RESYNC);
+
+  m_out_bl.clear();
+
+  std::string group_header_oid = librbd::util::group_header_name(
+      m_local_group_id);
+  m_in_flight_op_tracker.start_op();
+  auto aio_comp = create_rados_callback(
+    new LambdaContext([this](int r) {
+      handle_is_resync_requested(r);
+      m_in_flight_op_tracker.finish_op();
+    }));
+
+  int r = m_local_io_ctx.aio_operate(group_header_oid, aio_comp,
+                                     &op, &m_out_bl);
+  ceph_assert(r == 0);
+  aio_comp->release();
+}
+
+template <typename I>
+void Replayer<I>::handle_is_resync_requested(int r) {
+  std::unique_lock locker{m_lock};
+  dout(10) << "r=" << r << dendl;
+
+  if (is_replay_interrupted(&locker)) {
+    return;
+  }
+  std::string value;
+  if (r == 0) {
+    auto it = m_out_bl.cbegin();
+    r = librbd::cls_client::metadata_get_finish(&it, &value);
+  }
+  bool resync_requested = false;
+  if (r < 0 && r != -ENOENT) {
+    derr << "failed reading metadata: " << cpp_strerror(r) << dendl;
+  } else if (r == 0) {
+    resync_requested = true;
```

Comment:
Perhaps we could get rid of `resync_requested` variable altogether with something like
```
  if (r < 0 && r != -ENOENT) {
    derr << "failed reading metadata: " << cpp_strerror(r) << dendl;
  } else if (r == 0) {
    dout(10) << "local group resync requested" << dendl;
    if (is_group_primary(m_remote_group_snaps)) {
      handle_replay_complete(&locker, 0, "resync requested");
      return;
    }
    dout(10) << "cannot resync as remote is not primary" << dendl;
  }
```